src/pydash/numerical.py

```diff
@@ -40,10 +48,25 @@
     "zscore",
 )
 
+T = t.TypeVar("T")
+T2 = t.TypeVar("T2")
+T3 = t.TypeVar("T3")
+NumT = t.TypeVar("NumT", int, float)
```

Comment:
Should `Decimal` be added?

Is this different than `NumberT` or `NumberNoDecimalT`?

Reply:
Yes this is different, `NumberT` is only a `Union` of types, this `NumT` gets bound to one of the types in the signature of a function. Small example:
```python
import typing as t

NumberT = t.Union[int, float]
NumT = t.TypeVar("NumT", int, float)

def add(x: NumberT, y: NumberT) -> NumberT:
    return x + y

def add_generic(x: NumT, y: NumT) -> NumT:
    return x + y
    
reveal_type(add(5, 5))  # Revealed type is "Union[builtins.int, builtins.float]"
reveal_type(add_generic(5, 5))  # Revealed type is "builtins.int"
```
So here `NumberT` is always fixed to a union of types. But `NumT` gets bound to the detected type of `x` and `y` hence the return type is also `int`.

However this is not needed in most of the cases I used it in. I removed it and retyped `clamp` which I think I didn't type well.